airflow/models/taskinstance.py

```diff
@@ -237,7 +237,8 @@ def clear_task_instances(
         )
         for dr in drs:
             dr.state = dag_run_state
-            dr.start_date = timezone.utcnow()
+            dr.start_date = None
+            dr.last_scheduling_decision = None
```

Comment:
Oh, _technically_ someone could set a state other than queued here -- so maybe we need an `if dag_run_state == State.QUEUED: ... else: ...` here?

Reply:
I have set this and added test. Thanks!